# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: image/FrameAnimator.cpp
# Commit: 75b04de7e99c
# Full Hash: 75b04de7e99cab7bed5cc8b0f2f6d52ddf174bf3
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2019-05-25 09:39:13
# Regressor Bug: 1550422
# File Overlap Count: 1
# Description:
#   Bug 1550422 - P15. Move Skip and Once gfxPrefs to StaticPrefs. r=jrmuizel
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D31259
# ==============================================================================

diff -r 91c3acdb2454 -r 75b04de7e99c image/FrameAnimator.cpp
--- a/image/FrameAnimator.cpp	Fri May 24 11:32:20 2019 +0000
+++ b/image/FrameAnimator.cpp	Fri May 24 11:32:54 2019 +0000
@@ -7,10 +7,10 @@
 
 #include "mozilla/Move.h"
 #include "mozilla/CheckedInt.h"
+#include "mozilla/StaticPrefs.h"
 #include "imgIContainer.h"
 #include "LookupResult.h"
 #include "RasterImage.h"
-#include "gfxPrefs.h"
 
 namespace mozilla {
 
@@ -86,7 +86,7 @@
     } else if (aResult.Type() == MatchType::NOT_FOUND ||
                aResult.Type() == MatchType::PENDING) {
       if (mHasRequestedDecode) {
-        MOZ_ASSERT(gfxPrefs::ImageMemAnimatedDiscardable());
+        MOZ_ASSERT(StaticPrefs::ImageMemAnimatedDiscardable());
         mCompositedFrameInvalid = true;
       }
     }
@@ -379,7 +379,7 @@
   // only advance the frame if the current time is greater than or
   // equal to the current frame's end time.
   if (!currentFrame) {
-    MOZ_ASSERT(gfxPrefs::ImageMemAnimatedDiscardable());
+    MOZ_ASSERT(StaticPrefs::ImageMemAnimatedDiscardable());
     MOZ_ASSERT(aState.GetHasRequestedDecode() &&
                !aState.GetIsCurrentlyDecoded());
     MOZ_ASSERT(aState.mCompositedFrameInvalid);
@@ -451,7 +451,7 @@
       aMarkUsed);
 
   if (aState.mCompositedFrameInvalid) {
-    MOZ_ASSERT(gfxPrefs::ImageMemAnimatedDiscardable());
+    MOZ_ASSERT(StaticPrefs::ImageMemAnimatedDiscardable());
     MOZ_ASSERT(aState.GetHasRequestedDecode());
     MOZ_ASSERT(!aState.GetIsCurrentlyDecoded());
     if (result.Type() == MatchType::NOT_FOUND) {